### PR TITLE
#1841 Add tests for <LoadingSpinner> component

### LINKED ISSUE #1841: Add tests for <LoadingSpinner> component
Write unit tests for the `<LoadingSpinner>` React component to ensure expected behavior, edge case handling, and rendering logic.

## Essential Test Coverage Checklist

- [ ] **Renders successfully with minimal required props**  
- [ ] **Conditional rendering logic**  
- [ ] **Prop-based behavior** – different props affect output  
- [ ] **Event handling** – simulate user actions and verify callbacks  
- [ ] **State changes / internal logic**  
- [ ] **Default values and fallbacks**  
- [ ] **Text and content rendering**  
- [ ] **Handles edge cases and invalid inputs**  
- [ ] **Accessibility roles and labels**  
- [ ] **DOM structure / classNames / styles**

## Test Reference  
You can refer to the `AutoScrollToTop.test.tsx` file for an example of structure and best practices.  
To explore more examples, see the full component tests folder.

### PR DESCRIPTION
**Pull request by ParthGupta1304**
## Proposed change
Added a test file for the Loading Spinner React component.

Resolves #1841 

🧠 Overview
This PR introduces a complete set of unit tests for the LoadingSpinner component to improve the project's test coverage and reliability. The tests were written using Jest and React Testing Library in accordance with the Essential Test Coverage Checklist provided in the issue.

🔍 Test Scenarios Covered
Renders without crashing with minimal props
Handles invalid imageUrl input gracefully
Renders default fallback images for light/dark themes
Accepts and renders custom imageUrl when provided
Applies correct Tailwind CSS classes for theme modes
Ensures alt text for accessibility
Checks for expected DOM structure and animation class presence

🙏 Acknowledgements
Thanks to the OWASP/Nest team for the well-structured codebase and clear contribution guidelines. Happy to iterate if feedback is provided!

## Checklist

- [x] I've read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md).
- [ ] I've run `make check-test` locally; all checks and tests passed.

❗ Note: Unable to run make check-test due to Docker not being available locally. Instead, I ran all tests, lint checks, and type checks manually to ensure code correctness and consistency.

### CODE REVIEW

**Comment by kasya** [comment]
@ParthGupta1304 Could you resolve conflicts and run `make check` locally? Than push any updated file to this PR.

**Comment by ParthGupta1304** [comment]
Sure Kasya, I will work on it

**Comment by ParthGupta1304** [comment]
I have resolved the conflicts and ran the make check locally

File: frontend/package.json [hunk 1]
```diff
@@ -33,7 +34,7 @@
     "@next/eslint-plugin-next": "^15.4.3",
     "@next/third-parties": "^15.4.3",
     "@sentry/nextjs": "^9.41.0",
-    "@testing-library/user-event": "^14.6.1",
+    "@testing-library/dom": "^10.4.0",
```
Comment (kasya) [inline frontend/package.json]: Why did you make these updates? 🤔

File: package.json
Comment (kasya) [inline package.json:1]: Same question here. The task was to only create tests for the component 🤔

**Review by kasya — CHANGES_REQUESTED** [review]
hey @ParthGupta1304 Left some questions! Also there are conflicting files in this PR.

File: frontend/package.json (hunk 1 above)
Comment (ParthGupta1304) [reply frontend/package.json]: Hi kasya, I made these changes as these dependencies were required to test the changes

**Comment by ParthGupta1304** [comment]
Hi code reviewers, I have tried my best to resolve the conflict, but I think this is the best I could do for now.  I would really appreciate any suggestions and guidance since I am new to open source :).

**Review by arkid15r — DISMISSED** [review]
Please revert all unrelated changes. Keep frontend/__tests__/unit/components/LoadingSpinner.test.tsx file only.

**Comment by ParthGupta1304** [comment]
Hi code managers, I have reverted all the changes except the test file, and I can see no conflicts in the PR.
I request another code review, and I am happy to make required changes.
